src/api/kamino/transactions.ts

```
@@ -68,55 +69,119 @@ async function fetchTransactions(
+    transactions.push(
+      ...(await mapInstructionsToResponse(response.instructions, env, carpool.config, kamino, shareholder))
+    );
   }
   return { transactions, lastUpdatedOn: timestamp };
 }
 
 async function mapInstructionsToResponse(
   instructions: Instruction[],
   env: ENV,
-  config: HubbleConfig
+  config: HubbleConfig,
+  kamino: Kamino,
+  shareholder: PublicKey
 ): Promise<KaminoTransactionResponse[]> {
   const response: KaminoTransactionResponse[] = [];
+  const solMint = config.kamino.mints.find((x) => x.scopeToken === 'SOL')!.address;
   for (const instruction of instructions) {
     const transactionName = instruction.name;
     const transactionSignature = instruction.txSignature;
     const timestamp = new Date(instruction.timestamp);
-    const strategy = instruction.accounts.find((x) => x.name === 'strategy');
-    const userSharesAta = instruction.accounts.find((x) => x.name === 'userSharesAta');
-    const tokenAAta = instruction.accounts.find((x) => x.name === 'tokenAAta');
-    const tokenBAta = instruction.accounts.find((x) => x.name === 'tokenBAta');
-    if (!strategy || !userSharesAta || !tokenAAta || !tokenBAta) {
-      logger.warning(`Could not find correct accounts for transaction ${transactionSignature}`);
+    const strategyPubkey = instruction.accounts.find((x) => x.name === 'strategy');
+
+    if (!strategyPubkey) {
+      logger.error(`Could not find correct strategy account for transaction ${transactionSignature}`);
+      continue;
+    }
+
+    const redisKey = getKaminoStrategyRedisKey(strategyPubkey.pubKey, env);
+    const strategy = await redis.cacheFetchJson(
+      redisKey,
+      () => kamino.getStrategyByAddress(new PublicKey(strategyPubkey.pubKey)),
+      {
+        cacheExpiryType: CacheExpiryType.ExpireInSeconds,
+        cacheExpirySeconds: STRATEGIES_EXPIRY_IN_SECONDS,
+      }
+    );
+    if (!strategy) {
+      logger.error(`Could not find strategy ${strategyPubkey.pubKey}`);
```

no silent errors, this should throw, this is fatal